apps/prairielearn/src/pages/userSettings/userSettings.ts

```diff
@@ -73,17 +41,68 @@ router.get(
       });
     }
 
+    // Get all purchases for this user.
+    const allPurchases = await sqldb.queryRows(
+      sql.select_purchases,
+      { user_id: authn_user.user_id },
+      PurchaseRowSchema,
+    );
+
+    // Only show completed checkout Sessions. If the user clicks through to
+    // Stripe but never actually fills in their payment info and completes
+    // the checkout, we'll still have a session in the database but we don't
+    // want to show it to the user.
+    //
+    // Note that the status we check for here is independent of if the payment
+    // has actually come through; that's stored in the `payment_status` field.
+    const completedPurchases = allPurchases.filter(
+      (purchase) => purchase.stripe_checkout_session.data.status === 'complete',
+    );
```

Comment:
Maybe move this into the `ee` directory? We should certainly only do these reads if it's an enterprise instance.

Reply:
Keeping this code in `ee/` seems very reasonable.

Reply:
Moved this all to `ee` in bc26614.